app/services/update_query_from_params_service.rb

```diff
@@ -77,11 +79,19 @@ def apply_columns(params)
   end
 
   def apply_sums(params)
-    query.display_sums = params[:display_sums] if params[:display_sums]
+    query.display_sums = params[:display_sums] if params.key?(:display_sums)
   end
 
   def apply_timeline(params)
-    query.timeline_visible = params[:timeline_visible] if params[:timeline_visible]
+    query.timeline_visible = params[:timeline_visible] if params.key?(:timeline_visible)
+  end
+
+  def apply_hierarchy(params)
+    if query.grouped?
+      query.show_hierarchies = false
+    else
+      query.show_hierarchies = params[:show_hierarchies] if params.key?(:show_hierarchies)
+    end
```

Comment:
I am fine with the side effect as I can see why it is necessary. 

However, I'd then rather narrow down the cases when that side effect is applied. In addition, this method right now depends on the correct order of execution (only if `group_by` is set before calling the method, will `show_hierarchies` be ignored). In addition, this will prevent clients from getting an error response in a scenario where they only send `show_hierarchies=true` to a grouped query. The backend will currently simply discard the change. 

I'd therefore rather have one method which covers both params at the same time and which only takes the params as input.